pkg/systemd/notifyproxy/notifyproxy.go

```diff
@@ -88,15 +110,37 @@ func (p *NotifyProxy) WaitAndClose() error {
 	}()
 
 	for {
+		// Set a read deadline of one second such that we achieve a
+		// non-blocking read and can check if the container has already
+		// stopped running; in that case no READY message will be send
+		// and we're done.
+		if err := p.connection.SetReadDeadline(time.Now().Add(time.Second)); err != nil {
+			return err
+		}
 		buf := make([]byte, 1024)
 		num, err := p.connection.Read(buf)
+		if err != nil {
+			if !errors.Is(err, os.ErrDeadlineExceeded) {
+				return err
+			}
+		} else {
+			for _, s := range strings.Split(string(buf[:num]), "\n") {
+				if s == daemon.SdNotifyReady {
+					return nil
+				}
```

Comment:
This check seems fragile since we use a fixed buffer, what happens if the ready message starts at byte 1022? In this case the string will not match and the next read will also not match since the string was cut in two pieces.
Is there a guarantee that sd notify messages are always less then 1024 byte? Otherwise you would need to read until a newline or EOF.

Reply:
I think it doesn't hurt to make it more robust.  I found it to be sufficient since only the READY message is supported.